catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -377,6 +368,17 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 
 	public void setRotationMode(int mode) {
 		rotationMode = mode;
+		rotateLookDataIfNeeded(mode);
+	}
+
+	private void rotateLookDataIfNeeded(int mode) {
+		if (mode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && sprite.look.isFlipped()) {
+			getLookData().getTextureRegion().flip(true, false);
+		}
+		boolean orientedLeft = sprite.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (mode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			getLookData().getTextureRegion().flip(true, false);
+		}
```

Comment:
The first if in line 375 is responsible for the case, if a sprite had LR rotation before and got flipped at some point and while being flipped, the rotation style changed to something different than LR so it needs to be flipped to normal again.
The second if in 379 flips the lookdata if it's oriented left but not flipped yet.
One does exclude the other, if I get correctly what you meant to say, because of the "mode" part, only one if will ever be true so I could combine the ifs to make it smaller.